lib/logger/lib/logger/utils.ex

```diff
@@ -145,7 +145,7 @@ defmodule Logger.Utils do
     # arguments according to the truncate limit.
     {args, _} =
       Enum.map_reduce(args, truncate, fn arg, acc ->
-        if is_binary(arg) do
+        if is_binary(arg) and is_integer(acc) do
```

Comment:
Maybe we should have `acc != :infinity`? Otherwise someone may pass a completely invalid `acc` and we will silently discard it.

Reply:
I do a agree with that, I would feel silly if I introduced another unexpected behaviour